Add flyover to package level read me (#899)
Esri/arcgis-maps-sdk-kotlin-toolkit: toolkit/ar/README.md

View the API Reference for the AR module [here](https://developers.arcgis.com/kotlin/toolkit-api-reference/arcgis-maps-kotlin-toolkit/com.arcgismaps.toolkit.ar/index.html). 

The AR toolkit module provides three composable functions to render `ArcGISScene` content in AR:

| [TableTopSceneView](#TableTopSceneView)                         | [WorldScaleSceneView](#WorldScaleSceneView)                         | [FlyoverSceneView](#FlyoverSceneView)
|-----------------------------------------------------------------|---------------------------------------------------------------------|----------------------------------------|
| ![screenshot](screenshot.png)                                   | ![worldscale-screenshot](worldscale-screenshot.png)                 | ![flyover-screenshot](flyover-screenshot.png)

> | In tabletop, scene content is anchored to a real-world surface. | In world-scale AR, scene content is integrated with the real world. | In flyover the content is navigated by moving the device.

Author: Fixed

> | In tabletop, scene content is anchored to a real-world surface. | In world-scale AR, scene content is integrated with the real world. | On screen, flyover is visually indistinguishable from normal scene rendering. |

Reviewer: I'm not sure this is a useful tip here? Maybe better to comment on how scene content behaves as with the others?

```suggestion
| In tabletop, scene content is anchored to a real-world surface. | In world-scale AR, scene content is integrated with the real world. | In flyover AR, scene content at a remote location responds to device movements. |
```

Author: Yeah I changed it based on https://github.com/Esri/arcgis-maps-sdk-kotlin-toolkit/pull/899#discussion_r2168815117 but wasn't 100% comfortable with the result. 

I think your words are probably better.

Author: Fixed

To see it in action, check out the [microapp](../../microapps/ArWorldScaleApp).

## FlyoverSceneView

> The `FlyoverSceneView` provides an augmented reality fly over experience.

Reviewer 2: How's about adding a sentence:

> In flyover the content is navigated by moving the device, allowing the scene to be explored more naturally than with a touch or mouse interface.

The first half of this comes from the sentence I suggest removing from the table above, the second half from the introductory sentence in our [doc for the Java Flyover AR](https://github.com/Esri/arcgis-runtime-samples-android/tree/main/java/explore-scene-in-flyover-ar).

Author: Added

### Features

- A composable [FlyoverSceneView](../../toolkit/ar/src/main/java/com/arcgismaps/toolkit/ar/FlyoverSceneView.kt)
  that displays a `SceneView`.
- Uses device movement to position the camera viewing the scene.
- Provides parameters to configure and interact with the `SceneView`, such as specifying an
  `ArcGISScene` and a translation factor that defines how much the scene view translates as the
  device moves.
- A [FlyoverSceneViewProxy](../../toolkit/ar/src/main/java/com/arcgismaps/toolkit/ar/FlyoverSceneViewProxy.kt)
  is passed to the `FlyoverSceneView` composable function to define state such as the camera origin
  and perform operations such as identify.

> - A `FlyoverSceneViewScope` provided as the receiver by the `FlyoverSceneView`'s `content` lambda

Reviewer 2: Since we provide links to FlyoverSceneView & FlyoverSceneViewProxy above, also make FlyoverSceneViewScope a link for consistency?

Author: Fixed

See [this page](https://developers.google.com/ar/develop/privacy-requirements) for more information.

### Usage

The `FlyoverSceneView` requires camera permissions, which are requested by default when the
`FlyoverSceneView` enters composition. The following camera-related settings need to be specified in
the `AndroidManifest.xml`:

```xml
    <uses-permission android:name="android.permission.CAMERA" />

    <!-- Limits app visibility in the Google Play Store to ARCore supported devices
         (https://developers.google.com/ar/devices). -->
    <uses-feature android:name="android.hardware.camera.ar" />
    <uses-feature
        android:name="android.hardware.camera"
        android:required="true" />
```

If ARCore is not optional for your application to function (as is the case with
the [microapp](../../microapps/ArFlyoverApp)), you also need to add the following to your
`AndroidManifest.xml`:

```xml
        <!-- "AR Required" app, requires "Google Play Services for AR" (ARCore)
             to be installed, as the app does not include any non-AR features. -->
        <meta-data android:name="com.google.ar.core" android:value="required" />
```

> Configure an `ArcGISScene` with the data and base surface required for your table top use case:

Reviewer 2: table top -> flyover

Author: Fixed!

```kotlin
FlyoverSceneView(
    arcGISScene = arcGISScene,
    flyoverSceneViewProxy = flyoverSceneViewProxy,
    translationFactor = 1000.0,
    ...  
)
```

Pass an `onInitializationStatusChanged` callback to the `FlyoverSceneView` composable function to
get notified about initialization status changes:

```kotlin
FlyoverSceneView(
    arcGISScene = arcGISScene,
    flyoverSceneViewProxy = flyoverSceneViewProxy,
    translationFactor = 1000.0,
    onInitializationStatusChanged = { status ->
        updateStatus(status)
    }
  ...  
)
```

Make use of other features of a SceneView, for example handle `onSingleTapConfirmed` events and
display a `Callout` at the tapped location:

```kotlin
var tappedLocation by remember { 
    mutableStateOf<Point?>(null) 
}

FlyoverSceneView(
    arcGISScene = arcGISScene,
    flyoverSceneViewProxy = flyoverSceneViewProxy,
    translationFactor = 1000.0,
    onInitializationStatusChanged = { status ->
      updateStatus(status)
    },
    onSingleTapConfirmed = { singleTapConfirmedEvent ->
        tappedLocation = flyoverSceneViewProxy.screenToBaseSurface(singleTapConfirmedEvent.screenCoordinate)
    }
    ...
) {
    tappedLocation?.let {

>         Callout(location) {

Reviewer 2: Should this be `Callout(it)`?

```kotlin
@Composable
fun MainScreen() {
    ...
    val elevationSource =
        ArcGISTiledElevationSource("https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer")

    val meshLayer =
        IntegratedMeshLayer("https://tiles.arcgis.com/tiles/z2tnIkrLQ2BRzr6P/arcgis/rest/services/Girona_Spain/SceneServer")

    val arcGISScene by remember { 
        ArcGISScene(BasemapStyle.ArcGISImagery).apply { 
            baseSurface.elevationSources.add(elevationSource)
            operationalLayers.add(meshLayer)
        }
    }
    ...
}
```

Define a camera origin point and heading for your scene that serves as the starting point for
navigating:

```kotlin
val location = Point(
   2.82407,
   41.99101,
   230.0,
   SpatialReference.wgs84()
)
val heading = 160.0
```

Create a `FlyoverSceneViewProxy` with this state:

```kotlin
val flyoverSceneViewProxy = rememberFlyoverSceneViewProxy(location, heading)
```

Determine a translation factor that suits the scene and camera origin point. For example, a
relatively high translation factor is probably suitable when flying high above the scene, but a low

> value more suitable when exploring the scene at ground level. Call the `FlyoverSceneView`

Reviewer: ```suggestion
value may be more suitable when exploring the scene at ground level. Call the `FlyoverSceneView`
```

composable function

> with these parameters:

Reviewer: A bit nitpicky, but currently it sounds like you are saying this translation factor is most appropriate
```suggestion
specifying the parameters you chose:
```

Author: Fixed
